Sources/SolToBoogieTest/RegressionExecutor.cs

```diff
@@ -0,0 +1,177 @@
+            SolidityCompiler compiler = new SolidityCompiler();
+            CompilerOutput compilerOutput = compiler.Compile(solcPath, filePath);
+
+            if (compilerOutput.ContainsError())
+            {
+                compilerOutput.PrintErrorsToConsole();
+                throw new SystemException("Compilation Error");
+            }
+
+            // build the Solidity AST from solc output
+            AST solidityAST = new AST(compilerOutput, Path.GetDirectoryName(filePath));
+
+            // translate Solidity to Boogie
+            BoogieTranslator translator = new BoogieTranslator();
+            BoogieAST boogieAST = translator.Translate(solidityAST);
+
+            // dump the Boogie program to a file
+            using (var outWriter = new StreamWriter(outFile))
+            {
+                outWriter.WriteLine(boogieAST.GetRoot());
+            }
+
+            // read the corral configuration from Json
+            string configJsonPath = Path.Combine(configDirectory, Path.GetFileNameWithoutExtension(filename) + ".json");
+            string jsonString = File.ReadAllText(configJsonPath);
+            CorralConfiguration corralConfig = JsonConvert.DeserializeObject<CorralConfiguration>(jsonString);
+
+            string corralOutput = RunCorral(corralConfig);
+            return CompareCorralOutput(corralConfig.ExpectedResult, corralOutput);
+        }
+
+        private string RunCorral(CorralConfiguration corralConfig)
+        {
+            string corralArguments = GenerateCorralArguments(corralConfig);
+
+            Process p = new Process();
+            p.StartInfo.UseShellExecute = false;
+            p.StartInfo.RedirectStandardInput = false;
+            p.StartInfo.RedirectStandardOutput = true;
+            p.StartInfo.RedirectStandardError = true;
+            p.StartInfo.CreateNoWindow = true;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                p.StartInfo.FileName = corralPath;
+                p.StartInfo.Arguments = corralArguments;
+            }
+            else
+            {
+                p.StartInfo.FileName = "mono";
+                p.StartInfo.Arguments = $"{corralPath} {corralArguments}";
+                Console.WriteLine(p.StartInfo.Arguments);
+            }
+            p.Start();
+
+            string corralOutput = p.StandardOutput.ReadToEnd();
+            string errorMsg = p.StandardError.ReadToEnd();
+            if (!String.IsNullOrEmpty(errorMsg))
+            {
+                Console.WriteLine($"Error: {errorMsg}");
+            }
+            p.StandardOutput.Close();
+            p.StandardError.Close();
+
+            // TODO: should set up a timeout here
+            // but it seems there is a problem if we execute corral using mono
+
+            return corralOutput;
+        }
+
+        private bool CompareCorralOutput(string expected, string actual)
+        {
+            return actual.Contains(expected);

```

Comment:
Why contains and not equals?

Reply:
This method has been updated:
f91368327b6e66b116f1d93c835de5c9c2aeb7f6